### PR TITLE
access data using nested accessor keys

### PR DESCRIPTION
**Pull request by caravinci**
adds a function to recursively dig values out of nested data objects

### CODE REVIEW

File: packages/table-core/src/utils.tsx [hunk 1]
```diff
@@ -163,6 +163,27 @@ export function memo<TDeps extends readonly any[], TResult>(
   }
 }
 
+export function getProp(
+  dataObj: Record<string, any>,
+  accessorKeys: string | string[] = '',
+  defaultVal = ''
+): string {
```
Comment (caravinci) [inline packages/table-core/src/utils.tsx:170]: meant to replicate (at a rather basic level) lodash's `get`

File: packages/table-core/src/utils.tsx [hunk 2]
```diff
@@ -163,6 +163,27 @@ export function memo<TDeps extends readonly any[], TResult>(
...
+  const keys = Array.isArray(accessorKeys)
+    ? accessorKeys
+    : accessorKeys
+        .replace(/(\[(\d)\])/g, '.$2')
```
Comment (caravinci) [inline packages/table-core/src/utils.tsx:174]: this allows `accessor.[0].accessor' type of keys... WIP

File: packages/table-core/src/utils.tsx [hunk 3]
```diff
@@ -163,6 +163,27 @@ export function memo<TDeps extends readonly any[], TResult>(
...
+        .split('.')
+        .filter(key => !['', undefined].includes(key))
+
+  const value = dataObj[keys[0]]
+
+  if (value && keys.length > 1) {
+    return getProp(value, keys.slice(1), defaultVal)
```
Comment (caravinci) [inline packages/table-core/src/utils.tsx:181]: recursively keeps on digging until it find the innermost property

File: packages/table-core/src/utils.tsx [hunk 4]
```diff
@@ -163,6 +163,27 @@ export function memo<TDeps extends readonly any[], TResult>(
...
+  }
+
+  return value ?? defaultVal
```
Comment (caravinci) [inline packages/table-core/src/utils.tsx:184]: returning an empty string if nothing is found.

File: packages/table-core/src/core.tsx [hunk 5]
```diff
@@ -331,7 +332,7 @@ export function createTableInstance<TGenerics extends AnyGenerics>(
         accessorFn = columnDef.accessorFn
       } else if (columnDef.accessorKey) {
         accessorFn = (originalRow?: TGenerics['Row']) =>
-          (originalRow as any)[columnDef.accessorKey]
+          getProp(originalRow as any, columnDef.accessorKey)
```
Comment (caravinci) [inline packages/table-core/src/core.tsx:335]: get the nested property, rather than expecting data like `{ 'accessor.accessor': value }`

**Review by caravinci — COMMENTED** [review]
couple of comments to explain intention, and facilitate feedback

**Comment by tannerlinsley** [comment]
This would need full typesafety/autocomplete before I’ll consider merging. This means I would want to see “foo.bar.baz” auto complete.

**Comment by caravinci** [comment]
Absolutely fair! that'd be great, and I'd love to help... but I'm not quite sure how to implement something like that. Closing this so others can take a shot at it.

Thanks again for this amazing library!

**Comment by orenklein** [comment]
@caravinci You can take a look at this https://stackoverflow.com/a/58436959/15026946 for the needed type.
Not sure how it might affect performance though.

**Pull request closed without merging**